packages/babel-core/src/config/options.js

```diff
@@ -0,0 +1,255 @@
+// @flow
+
+import removed from "./removed";
+import {
+  assertString,
+  assertBoolean,
+  assertObject,
+  assertInputSourceMap,
+  assertIgnoreList,
+  assertPluginList,
+  assertFunction,
+  assertSourceMaps,
+  assertCompact,
+  assertSourceType,
+} from "./option-assertions";
+
+type ValidatorSet = {
+  [string]: Validator<any>,
+};
+
+type Validator<T> = (string, mixed) => T;
+
+const ROOT_VALIDATORS: ValidatorSet = {
+  filename: (assertString: Validator<
+    $PropertyType<ValidatedOptions, "filename">,
+  >),
```

Comment:
...what is this syntax?

It looks like an incomplete arrow function declaration to me (missing the `=>` and return type)

Reply:
I basically did this as a nice way to make sure that the validator function actually validates the type that we declared in `ValidatedOptions`. So this is a case that cases the validator function into a function that validates the type declared in `ValidatedOptions`. If the cast fails, we know that something has gone wrong.